sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -74,7 +77,19 @@ abstract class ExternalCatalog {
    */
   def alterDatabase(dbDefinition: CatalogDatabase): Unit
 
-  def getDatabase(db: String): CatalogDatabase
+  def getDatabase(db: String): CatalogDatabase = {
+    val database = getDatabaseInternal(db)
+    // The default database's location always uses the warehouse path.
+    // Since the location of database stored in metastore is qualified,
+    // we also make the warehouse location qualified.
+    if (db == SessionCatalog.DEFAULT_DATABASE) {
```

Comment:
I think if we create an default  instance  of `CatalogDatabase`, we will add more logic to other interfaces, such as `databaseExists` ,we need to add the logic that  if it is default database, we always return true even if there is no default database in hive.